.github/workflows/publish-test-utils-image.yaml

```diff
@@ -17,43 +17,39 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
 
+permissions: {}
+
 jobs:
   test-utils:
     runs-on: ubuntu-20.04
 
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          persist-credentials: false
 
       - name: Docker meta
         id: meta
-        uses: docker/metadata-action@v5
+        uses: docker/metadata-action@902fa8ec7d6ecbf8d84d538b9b233a880e428804 # v5.7.0
         with:
           images: ghcr.io/${{ github.repository }}/test-utils
 
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
+        uses: docker/setup-qemu-action@29109295f81e9208d7d86ff1c6c12d2833863392 # v3.6.0
 
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-
-      - name: Cache Docker layers
```

Comment:
Why was this step removed, is this caching step a security issue?

Reply:
Maybe because it's in the publish GitHub action, in that case, we should also remove `cache-from` and `cache-to` from the `docker/build-push-action` action.

Reply:
Yes, as Suraj pointed out the cache action is vulnerable to [cache poisoning](https://woodruffw.github.io/zizmor/audits/#cache-poisoning). I figured it wasn't executed that frequently that justified keeping it.

>we should also remove cache-from and cache-to from the docker/build-push-action action.

Yes, nice catch.